model/common/src/icon4py/model/common/orchestration/decorator.py

```diff
@@ -0,0 +1,584 @@
+
+
+try:
+    import dace
+except ImportError:
+    dace: Optional[ModuleType] = None  # type: ignore[no-redef]
+
+try:
+    import ghex
+    from ghex import expose_cpp_ptr
+except ImportError:
+    ghex: Optional[ModuleType] = None  # type: ignore[no-redef]
+
+
+def dace_orchestration() -> bool:
+    """DaCe Orchestration: GT4Py Programs called in a dace.program annotated function"""
+    if settings.dace_orchestration is not None:
+        return True
+    return False
+
+
+def orchestration(method=True):
+    def decorator(fuse_func):
+        compiled_sdfgs = {}  # Caching
+
+        def wrapper(*args, **kwargs):
+            if dace_orchestration():
+                if "dace" not in settings.backend.executor.name.lower():
+                    raise ValueError(
+                        "DaCe Orchestration works only with DaCe backends. Change the backend to a DaCe supported one."
+                    )
+
+                if method:
+                    # self is used to retrieve the _exchange object -on the fly halo exchanges- and the grid object -offset providers-
+                    self = args[0]
+                    self_name = next(
+                        param.name for param in inspect.signature(fuse_func).parameters.values()
+                    )
+                else:
+                    raise ValueError(
+                        "The orchestration decorator is only for methods -at least for now-."
+                    )
+
+                fuse_func_orig_annotations = copy.deepcopy(fuse_func.__annotations__)
+                fuse_func.__annotations__ = to_dace_annotations(
+                    fuse_func
+                )  # every arg/kwarg is annotated with DaCe data types
+
+                has_exchange = False
+                has_grid = False
+                for attr_name, attr_value in self.__dict__.items():
+                    if isinstance(attr_value, decomposition.ExchangeRuntime) and not has_exchange:
+                        has_exchange = True
+                        exchange_obj = getattr(self, attr_name)
+                    if isinstance(attr_value, icon_grid.IconGrid) and not has_grid:
+                        has_grid = True
+                        grid = getattr(self, attr_name)
+
+                if not has_exchange or not has_grid:
+                    raise ValueError("No exchnage/grid object found.")
+
+                compile_time_args_kwargs = {}
+                all_args_kwargs = list(args) + list(kwargs.values())
+                for i, (k, v) in enumerate(fuse_func.__annotations__.items()):
+                    if v is dace.compiletime:
+                        compile_time_args_kwargs[k] = all_args_kwargs[i]
+                unique_id = hash(tuple([id(e) for e in compile_time_args_kwargs.values()]))
+
+                default_build_folder = os.path.join(
+                    ".dacecache", f"MPI_rank_{exchange_obj.my_rank()}"
```

Comment:
Is there a way to get ".dacecahe" through the dace API? like `dace.config.Config().get("default_build_folder")`